src/routes/safe/components/Transactions/TxList/TxQueueRow.tsx

```diff
@@ -15,15 +22,21 @@ type TxQueueRowProps = {
 export const TxQueueRow = ({ isGrouped = false, transaction }: TxQueueRowProps): ReactElement => {
   const { activeHover } = useContext(TxHoverContext)
   const [tx, setTx] = useState<Transaction>(transaction)
+  const willBeReplaced = tx.txStatus === LocalTransactionStatus.WILL_BE_REPLACED ? ' will-be-replaced' : ''
+  const isPending = useSelector((state: AppReduxState) => isTxPending(state, transaction.id))
+  const pendingTx = useSelector(pendingTxByChain)
+  const pendingTxNonce = isMultisigExecutionInfo(pendingTx?.executionInfo) ? pendingTx?.executionInfo.nonce : undefined
+  const nonce = isMultisigExecutionInfo(transaction.executionInfo) ? transaction.executionInfo.nonce : undefined
+  const isReplacementTxPending = !isPending && nonce && pendingTxNonce && nonce === pendingTxNonce
 
   useEffect(() => {
-    if (activeHover && activeHover !== transaction.id) {
+    if ((activeHover && activeHover !== transaction.id) || isReplacementTxPending) {
       setTx((currTx) => ({ ...currTx, txStatus: LocalTransactionStatus.WILL_BE_REPLACED }))
       return
     }
 
     setTx(transaction)
-  }, [activeHover, transaction])
+  }, [activeHover, transaction, isPending, nonce, pendingTxNonce])
```

Comment:
You need to add isReplacementTxPending as a dependancy.

Reply:
Good catch! Fixed it